test/libraries/MultiSend.spec.ts

```diff
@@ -174,5 +175,63 @@ describe("MultiSend", async () => {
                 ),
             ).to.be.eq("0x" + "baddad".padEnd(64, "0"));
         });
+
+        it("can bubble up revert message on call", async () => {
+            const { safe, multiSend, mock } = await setupTests();
+
+            const user2Safe = safe.connect(user2);
+            await executeContractCallWithSigners(safe, safe, "enableModule", [user2.address], [user1]);
+            await mock.givenCalldataRevertWithMessage("0xbaddad", "Some random message");
+
+            const txs: MetaTransaction[] = [
+                buildSafeTransaction(
+                    Object.assign({
+                        to: mock.address,
+                        data: "0xbaddad",
+                        operation: 0,
+                    }),
+                ),
+            ];
+            const { data } = buildMultiSendSafeTx(multiSend, txs, await safe.nonce());
+
+            await expect(await user2Safe.callStatic.execTransactionFromModuleReturnData(multiSend.address, 0, data, 1)).to.be.deep.eq([
+                false,
+                "0x08c379a000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000013536f6d652072616e646f6d206d65737361676500000000000000000000000000",
+            ]);
+        });
+
+        it("can bubble up revert message on delegatecall", async () => {
+            const { safe, multiSend, mock } = await setupTests();
+
+            const user2Safe = safe.connect(user2);
+            await executeContractCallWithSigners(safe, safe, "enableModule", [user2.address], [user1]);
+            const { data: setRevertMessageData } = await mock.populateTransaction.givenCalldataRevertWithMessage(
+                "0xbaddad",
+                "Some random message",
+            );
+
+            const txs: MetaTransaction[] = [
+                buildSafeTransaction(
+                    Object.assign({
+                        to: mock.address,
+                        data: setRevertMessageData,
+                        operation: 1,
+                    }),
+                ),
+                buildSafeTransaction(
+                    Object.assign({
+                        to: mock.address,
+                        data: "0xbaddad",
+                        operation: 1,
+                    }),
+                ),
+            ];
+            const { data } = buildMultiSendSafeTx(multiSend, txs, await safe.nonce());
+
+            await expect(await user2Safe.callStatic.execTransactionFromModuleReturnData(multiSend.address, 0, data, 1)).to.be.deep.eq([
```

Comment:
There's a chai matcher that checks for revert: https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#revert-with-message

In the current form, it checks the return data equality and not the fact it was reverted

Reply:
the 'execTransactionFromModuleReturnData' in 'moduleManager' does not revert if the call fails it succeeds and returns false and the bytes memory

I used this path because it is the only one that relays the error message from the call to outside the safe

Reply:
You could use one of the available hardhat EOA accounts in this case.

Reply:
Makes sense,

But the multSend library would still need to receive an delegateCall so I would use another mock for that,

what do you think about keeping these new tests as integration tests and I write some more 'unit' test like?

Reply:
The question is, what do those integration tests actually test? That `execTransactionFromModuleReturnData` returns data? I believe we already have a test for that... Given that this is a library contract and the revert data cannot be used by a Safe core contract, I believe the value of such a test is very little. But I'd be keen to hear other's opinions as well; cc @rmeissner @akshay-ap

Reply:
Hello,

About this test, what is your preferred way to move foward?

An alternative I see is to write a mock to make delegatecalls and it calls the multSend libraries. Would that be acceped?

Reply:
Yes, this alternative sounds good to me

Reply:
I've sent new versions of the tests, let me know your remarks